src/StorageWriter/StorageWriter.ts

```diff
@@ -13,91 +13,121 @@ import { DAOClaims, DAOClaimsConfiguration } from './DAOClaims'
 import { DAOIntegrityCheckFailures } from './DAOIntegrityCheckFailures'
 import { ExchangeConfiguration } from './ExchangeConfiguration'
 import { Router } from './Router'
-import { Service, ServiceConfiguration } from './Service'
+import * as Service from './Service'
 
 export interface StorageWriterConfiguration
   extends LoggingConfiguration,
-    ServiceConfiguration,
+    Service.Configuration,
     DAOClaimsConfiguration {
   readonly ipfs: IPFSConfiguration
   readonly dbUrl: string
   readonly rabbitmqUrl: string
   readonly exchanges: ExchangeConfiguration
 }
 
-@injectable()
-export class StorageWriter {
-  private readonly logger: Pino.Logger
-  private readonly configuration: StorageWriterConfiguration
-  private readonly container = new Container()
-  private mongoClient: MongoClient
-  private dbConnection: Db
-  private integrityCheckFailuresCollection: Collection
-  private router: Router
-  private messaging: Messaging
-  private service: Service
-  private daoClaims: DAOClaims
-  private daoIntegrityCheckFailures: DAOIntegrityCheckFailures
-
-  constructor(configuration: StorageWriterConfiguration) {
-    this.configuration = configuration
-    this.logger = createModuleLogger(configuration, __dirname)
-  }
+export interface Dependencies {
+  readonly logger: Pino.Logger
+}
 
-  async start() {
-    this.logger.info({ configuration: this.configuration }, 'StorageWriter Starting')
-    this.mongoClient = await MongoClient.connect(this.configuration.dbUrl)
-    const db = this.dbConnection = await this.mongoClient.db()
+export interface FactoryArg {
```

Comment:
Swore I deleted this because its not used, but its back.. same with dependencies.